modules/cluster/main.tf

```diff
@@ -54,6 +54,31 @@ module "vpc" {
   }
 }
 
+// ----------------------------------------------------------------------------
+// Manually Tag VPC Subnets if needed
+// See https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html#vpc-subnet-tagging
+// ----------------------------------------------------------------------------
+resource "aws_ec2_tag" "public_subnet_tag" {
```

Comment:
Need to think about it a bit more, if someone is not creating vpc using this module, they can always use https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/variables.tf#L427-L438 to tag the subnets, right? In that case, it might lead to issues.
```
This tagging resource should not be combined with the Terraform resource for managing the parent resource. For example, using aws_vpc and aws_ec2_tag to manage tags of the same VPC will cause a perpetual difference where the aws_vpc resource will try to remove the tag being added by the aws_ec2_tag resource.
```
Source: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_tag
See also: https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1283

Reply:
I believe that's what the `ignore_tags` block achieves. See https://stackoverflow.com/a/64938829. I'm not all too familiar with Terraform though